### ISSUE TITLE
Commit only if Pending Commit

### ISSUE THREAD

**Issue by RyanMBess** [PaloAltoNetworks/pan-os-ansible#356]
The commit module found here https://paloaltonetworks.github.io/pan-os-ansible/modules/panos_commit_firewall_module.html commits even if there isn't any pending commit.  Please build in support that the module will check if there's a pending commit for the "admins".  If there are no pending commits for that admin, don't commit.  If there IS a pending commit for that admin, then commit.

**Comment by jamesholland-uk** [comment]
Hi @RyanMBess, can I clarify my understanding of the scenario please?

Example: Admin1 makes a change, and does not commit. Admin2 has made no changes.
- If admin1 sends a commit via Ansible, commit should execute because admin1 has uncommitted changes.
- If admin2 sends a commit via Ansible, commit should not execute, because admin2 does not have uncommitted changes (only admin1 has uncommitted changes).

Is that the use case you're requesting? Thanks!

**Comment by ai-ryanbess** [comment]
Hey James,
When running the palo ansible modules under an account called "Admin1".  Lets say we want to maintain address objects by having the source of truth be the CMDB.  So the code runs under "Admin1" pulls the data from the CMDB and validates that it matches what is in the Palo.  The last task in this playbook is to commit and thus in this instance a commit command is still issued even though there was nothing changed.  My solution to this was to create a handler that would call the next task only if a change occurred.  This has been my solution and likely what you would propose.  Thus given the same example if the CMDB and Palo matched, the handler wouldn't be called.  If the CMDB and Palo did not match thus there is a change, the handler would get called and would commit.

**Comment by RyanMBess** [comment]
Hey James,
When running the Palo ansible modules under an account called "Admin1". Lets say we want to maintain address objects by having the source of truth be the CMDB. So the code runs under "Admin1" pulls the data from the CMDB and validates that it matches what is in the Palo. The last task in this playbook is to commit and thus in this instance a commit command is still issued even though there was nothing changed. My solution to this was to create a handler that would call the next task only if a change occurred. This has been my solution and likely what you would propose. Thus given the same example if the CMDB and Palo matched, the handler wouldn't be called. If the CMDB and Palo did not match thus there is a change, the handler would get called and would commit.

**Comment by jamesholland-uk** [comment]
Hi @RyanMBess, if there is a task in a playbook executing ```panos_commit_firewall```, and there are no changes to commit, the task itself executes, but the commit which Ansible requests PAN-OS to execute should not actually happen. Do you observe this, or do you see different behaviour?

Example code:
```
    - name: Commit candidate configuration
      paloaltonetworks.panos.panos_commit_firewall:
        provider: "{{ device }}"
      register: results
    - name: Output results
      ansible.builtin.debug:
        msg: "Commit with Job ID: {{ results.jobid }} had output: {{ results }}"
```

Example when there are changes to commit:
```
TASK [Commit candidate configuration] ****************************************************************************************************************************
changed: [host_labfw]

TASK [Output results] ********************************************************************************************************************************************
ok: [host_labfw] => {
    "msg": "Commit with Job ID: 2782 had output: {'changed': True, 'jobid': '2782', 'details': ['Configuration committed successfully'], 'failed': False}"
}

PLAY RECAP *******************************************************************************************************************************************************
host_labfw                 : ok=3    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Example when there are no changes to commit:
```
TASK [Commit candidate configuration] ****************************************************************************************************************************
ok: [host_labfw]

TASK [Output results] ********************************************************************************************************************************************
ok: [host_labfw] => {
    "msg": "Commit with Job ID: 0 had output: {'changed': False, 'jobid': 0, 'failed': False}"
}

PLAY RECAP *******************************************************************************************************************************************************
host_labfw                 : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

**Comment by RyanMBess** [comment]
If i recall, i think this request came from the inability to push changes to firewalls at the admin level.  I see this is now fixed in 2.13.  I think we can close this request.

**Comment by jamesholland-uk** [comment]
No problem, thanks for the feedback
